modules/swagger-jaxrs/src/main/java/com/wordnik/swagger/jaxrs/Reader.java

```diff
@@ -70,9 +102,31 @@ public Reader(Swagger swagger, ReaderConfig config) {
     this.config = new DefaultReaderConfig(config);
   }
 
-  public Swagger read(Set<Class<?>> classes) {
+  public synchronized Swagger read(Set<Class<?>> classes) {
+
+    listeners.clear();
+
+    for(Class<?> cls: classes) {
+      if( ReaderListener.class.isAssignableFrom( cls ) && listeners.containsKey( cls )){
```

Comment:
Second condition still lacks for '!' operator.
Do you think it's safe to instantiate REST related resources? Please take a look at the resource from the #1085. To add a listener for the TestResource class the author will need to rework his/her class and add a default constructor.